osu.Game/Screens/Play/Player.cs

```diff
@@ -130,6 +131,11 @@ public abstract class Player : ScreenWithBeatmapBackground, ISamplePlaybackDisab
         /// </summary>
         protected virtual bool CheckModsAllowFailure() => GameplayState.Mods.OfType<IApplicableFailOverride>().All(m => m.PerformFail());
 
+        /// <summary>
+        /// Custom hotkeys which will be added to the <see cref="RulesetSkinProvidingContainer"/>.
+        /// </summary>
+        protected virtual IEnumerable<Container> GetCustomRulesetHotkeys() => Array.Empty<Container>();
```

Comment:
This shouldn't be called "RulesetHotkeys". It has nothing to do with rulesets.

Also, we'd sooner expose a target container than make a creat method like this. I'm not 100% sure why you are required to add inside it, but you could already use `DrawableRuleset` or another option. We wouldn't expose a function just for only the editor to use.